Userland/Libraries/LibCore/Stream.cpp

```diff
@@ -165,6 +165,31 @@ ErrorOr<size_t> File::read(Bytes buffer)
     return nread;
 }
 
+ErrorOr<ByteBuffer> File::read_all()
+{
+    off_t file_size = 0;
+    struct stat st;
+    int rc = fstat(m_fd, &st);
+    if (rc == 0)
+        file_size = st.st_size;
```

Comment:
This will not work for files in `/proc` which have a `st_size` of zero and `stdin`. Avoid this and use EOF as exit condition instead. If this is only meant as a heuristic then it's okay, but then please use `Core::System` wrappers.